apps/web/scripts/kernel/handlers/solanaHandler.ts

```diff
@@ -37,47 +38,126 @@ export const handleSignMessage: MessengerCallback = async (
 	payload,
 	channel,
 ) => {
-	const privateKey = await triggerActionToGetPrivateKey();
-
-	if (!privateKey) {
-		return;
-	}
-
-	const message = decode(payload.message);
-	const signature = signMessage(message, privateKey);
-
-	channel.postMessage({
-		from: 'walless@kernel',
-		requestId: payload.requestId,
-		signature: encode(signature),
-	});
-
-	return signature;
+	const { requestId = '' } = payload;
+	const popup = await handleOpenPopup(PopupType.SIGNATURE_POPUP, requestId);
+	requestMap[requestId] = {
+		channel,
+		payload,
+		popup,
+		resolve: false,
+	};
 };
 
-export const handleSignTransaction: MessengerCallback = async (
+export const handleResolveRequestSignature: MessengerCallback = async (
```

Comment:
`signature` is just the thing `kernel` must response to `sdk`, under some action to generate this signature. The name could be meaningful